openedx/license-manager
# ENT-2749: Enabling License creation from Subscription 'add' form

## Description

A field has been added to the 'add' form for creating subscriptions to specify the number of licenses desired for association with the given subscription. 

The is_active field has been added to the SubscriptionPlan model.

All other fields have been made read-only on the update screen for a SubscriptionPlan.

Link to the associated ticket: https://openedx.atlassian.net/browse/ENT-2749

## Testing considerations

- Include instructions for any required manual tests, and any manual testing that has
already been performed.
- Include unit and a11y tests as appropriate
- Consider performance issues.
- Check that Database migrations are backwards-compatible

## Testing instructions

- Checkout branch locally, run `make app-shell`, run `./manage.py migrate`
- Authenticate in LMS, navigate to license-manager Django Admin: http://localhost:18170/admin
- Create a SubscriptionPlan: http://localhost:18170/admin/subscriptions/subscriptionplan/add/
- Observe that the appropriate number of licenses have been created: http://localhost:18170/admin/subscriptions/license/
- Observe that the appropriate fields for the created SubscriptionPlan are now read-only: http://localhost:18170/admin/subscriptions/subscriptionplan/<uuid>/change/
- Try to decrease the number of Licenses associated with the SubscriptionPlan (field error should be raised)
- Increase the number of Licenses associated with the SubscriptionPlan and observe that the appropriate number of Licenses now exist for the associated SubscriptionPlan

## Post-review

Squash commits into discrete sets of changes
